### PR TITLE
Issue #SB-25459 fix: Report failing because of one of the datasets is failed to load

### PR DESCRIPTION
**Pull request by rakeshSgr**
…ort has no data

### CODE REVIEW

File: src/app/client/src/app/modules/dashboard/components/report/report.component.ts [hunk 1]
```diff
@@ -155,8 +155,9 @@ export class ReportComponent implements OnInit {
               map((apiResponse) => {
                 const [data, reportSummary] = apiResponse;
                 const result: any = Object.assign({});
-                const chart = (charts && this.reportService.prepareChartData(charts, data, updatedDataSource,
+                let chart = (charts && this.reportService.prepareChartData(charts, data, updatedDataSource,
```
Comment (yravinderkumar33) [inline src/app/client/src/app/modules/dashboard/components/report/report.component.ts:158]: move this logic inside prepareChartData method only..

File: src/app/client/src/app/modules/dashboard/services/report/report.service.ts [hunk 2]
```diff
@@ -53,20 +56,19 @@ export class ReportService {
...
+
+        response = response.filter(function(item){ if(item ){ return item.loaded = true } });
         return this.getFileMetaData(dataSources).pipe(
           map(metadata => {
             return _.map(response, res => {
               res.lastModifiedOn = _.get(metadata[res.id], 'lastModified');
               return res;
             });
           })
+          ,catchError(err => {
```
Comment (yravinderkumar33) [inline src/app/client/src/app/modules/dashboard/services/report/report.service.ts:68]: Remove this code block if you're not doing anything with the error and rethrowing it again.

File: src/app/client/src/app/modules/dashboard/components/report/report.component.ts (hunk 1 above)
Comment (rakeshSgr) [reply src/app/client/src/app/modules/dashboard/components/report/report.component.ts]: resolved

File: src/app/client/src/app/modules/dashboard/services/report/report.service.ts (hunk 2 above)
Comment (rakeshSgr) [reply src/app/client/src/app/modules/dashboard/services/report/report.service.ts]: removed

**Review by rajeevsathish — APPROVED** [review]

**Pull request merged**